sys/kern/sys_process.c

```diff
@@ -717,39 +725,43 @@ sys_ptrace(struct thread *td, struct ptrace_args *uap)
 			error = copyin(uap->addr, &r.ptevents, uap->data);
 		break;
 	case PT_IO:
-		error = COPYIN(uap->addr, &r.piod, sizeof r.piod);
+		error = copyincap(uap->addr, &r.piod, sizeof r.piod);
 		break;
 	case PT_VM_ENTRY:
 #if __has_feature(capabilities)
-	{
-		struct ptrace_vm_entry pve;
-		error = COPYIN(uap->addr, &pve, sizeof pve);
-		if (error)
-			break;
-
-		r.pve.pve_entry     = pve.pve_entry;
-		r.pve.pve_timestamp = pve.pve_timestamp;
-		r.pve.pve_start     = pve.pve_start;
-		r.pve.pve_end       = pve.pve_end;
-		r.pve.pve_offset    = pve.pve_offset;
-		r.pve.pve_prot      = pve.pve_prot;
-		r.pve.pve_pathlen   = pve.pve_pathlen;
-		r.pve.pve_fileid    = pve.pve_fileid;
-		r.pve.pve_fsid      = pve.pve_fsid;
-		r.pve.pve_path      = (void * __capability)(intcap_t)pve.pve_path;
-	}
+		error = copyincap(uap->addr, (char *)&r.pve, sizeof r.pve);
```

Comment:
It is better to check this.
I tested this change using fasttrap (that works), but PT_VM_ENTRY is never used there